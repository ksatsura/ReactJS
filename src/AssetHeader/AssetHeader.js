//imports
import React from 'react';
import AssetInfo from '../AssetInfo/AssetInfo';
import Poster from '../Poster/Poster';
import { connect } from 'react-redux';
import { backToSearch } from '../actions/assetActions';
import CompanyTitle from '../CompanyTitle/CompanyTitle';
import '../style.css';

export const AssetHeader = (props) => {

    const {asset, handleClick} = props;

    return (
        <div className='asset-info'>
            <div className='top-asset-info'>
                <CompanyTitle />
                <button onClick={handleClick} className='search-navigation'>search</button>
            </div>
            <div className='asset-poster-data'>
                <div className='poster'>
                    <Poster url={asset.poster_path}/>
                </div>
                <AssetInfo asset={asset} />
            </div>
        </div>
    );
};

const mapStateToProps = state => {

    return {
        asset: state.assetReducer.asset,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        handleClick: () => { dispatch(backToSearch()); },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(AssetHeader);