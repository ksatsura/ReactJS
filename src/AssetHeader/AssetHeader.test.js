import React from 'react';
import { AssetHeader } from './AssetHeader';
import {shallow} from 'enzyme';

describe('AssetHeader.js', () => {

    const props = {
        asset: {
            poster_path: ''
        },
        handleClick: jest.fn(),
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<AssetHeader {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<AssetHeader {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('handleClick', () => {

        it('test search button click event', () => {
        
            const component = shallow(<AssetHeader {...props} />);
    
            component.find('button').simulate('click');
            expect(props.handleClick.mock.calls.length).toEqual(1);
        });

    });
});
