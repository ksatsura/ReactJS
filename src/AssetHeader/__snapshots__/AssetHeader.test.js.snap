// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetHeader.js renders renders for desktop 1`] = `ShallowWrapper {}`;

exports[`AssetHeader.js renders renders for mobile 1`] = `ShallowWrapper {}`;
